Add WooCommerce header to WooCommerce.com requests

bgrgicak opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->
We want to add WooCommerce `user-agent` headers to WooCommerce.com API requests.

### How to test the changes in this Pull Request:


#### Test Extensions

1. Open WooCommerce > Extensions
2. Confirm that Featured extensions, categories, and search are working

#### Test My subscriptions
1. Open WooCommerce > Extensions
2. Open the _My Subscriptions _ tab
3. Connect to WooCommerce.com
4. Update your list of subscriptions
5. Confirm that your subscriptions are loaded

#### Test Status report
1. Open WooCommerce > Status
2. Scroll down to _Remote get_
3. Confirm that the field is loaded

#### Test inbox
1. Delete inbox notifications `DELETE FROM wp_customer_wc_admin_notes`
2. Delete the `wc_remote_inbox_notifications_specs wc_remote_inbox_notifications_stored_state` option
3. Run the `wc_admin_daily` cron: `wp cron event run wc_admin_daily`
4. Open [the home screen](https://wp.test:4443/wp-admin/admin.php?page=wc-admin)
6. Confirm that notifications are loaded

#### Test onboarding
1. Start the onboarding flow
2. On the first step chekc the Mailchimp signup
3. Confirm that all steps are loading correctly and that you can complete the onboarding wizard 


### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

bgrgicak committed:
Add WooCommerce header to WooCommerce.com requests

bgrgicak committed:
Remove duplicate user-agent

peterfabian reviewed (commented):
Thanks, looks good and tests well. 
I was wondering if you would like to also update the request over here: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/marketplace-suggestions/class-wc-marketplace-updater.php#L47

Or is that not used anymore?

Also, can you please rebase to resolve the conflict? Thanks!

bgrgicak committed:
Add header to suggestions request

bgrgicak committed:
Merge branch 'trunk' into add/wccom-api-headers

bgrgicak committed:
Add missing comma

bgrgicak commented:
> I was wondering if you would like to also update the request over here: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/marketplace-suggestions/class-wc-marketplace-updater.php#L47

Thanks, I missed this one. I added it now and fixed the conflict.

peterfabian reviewed (approved):
Looks good now, thanks!
